homeassistant/components/egps/__init__.py

```diff
@@ -0,0 +1,42 @@
+"""Energenie Power-Strip (EGPS) integration."""
+
+from pyegps import PowerStripUSB, get_device
+from pyegps.exceptions import MissingLibrary, UsbError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, IntegrationError
+
+from .const import CONF_DEVICE_API_ID, DOMAIN
+
+PLATFORMS = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Energenie Powerstrip."""
+    try:
+        powerstrip: PowerStripUSB | None = get_device(entry.data[CONF_DEVICE_API_ID])
+
+    except (MissingLibrary, UsbError) as ex:
+        raise IntegrationError("Can't access usb devices.") from ex
```

Comment:
Never seen this error, what does it do exactly and does HA handle that nicely in the UI? Did you consider ConfigEntryError

Reply:
I think that's best for now yes.